cpp/ql/src/Likely Bugs/Arithmetic/SignedOverflowCheck.ql

```diff
@@ -0,0 +1,25 @@
+/**
+ * @name Undefined result of signed test for overflow
+ * @description Testing for overflow by adding a value to a variable
+ *              to see if it "wraps around" works only for
+ *              `unsigned` integer values.
+ * @kind problem
+ * @problem.severity warning
+ * @precision high
+ * @id cpp/signed-overflow-check
+ * @tags reliability
+ *       security
+ */
+
+import cpp
+import semmle.code.cpp.valuenumbering.GlobalValueNumbering
+
+from RelationalOperation ro, AddExpr add, VariableAccess va1, VariableAccess va2
+where
+  ro.getAnOperand() = add and
+  add.getAnOperand() = va1 and
+  ro.getAnOperand() = va2 and
+  globalValueNumber(va1) = globalValueNumber(va2) and
+  add.getFullyConverted().getType().getUnspecifiedType().(IntegralType).isSigned() and
```

Comment:
I don't think we want `getFullyConverted` here in the signedness check. We want to check whether the addition is signed, not whether the result is converted to signed. That might fix the FP that Geoffrey mentioned today.

Reply:
Yes, that did the trick.  Also needed to check that the other side of the comparison was also signed (and not cast to an unsigned type).